hashtagsv2/graphs/views.py

```diff
@@ -0,0 +1,221 @@
+            earliest_date = earliest_date + relativedelta(months= +1)
+    # Split by years
+    else:
+        qs = hashtags.annotate(year = TruncYear('timestamp')).values('year').annotate(edits = Count('rc_id')).order_by()       
+        for item in qs:
+            time_dic[item['year'].date()] = item['edits']
+          
+        earliest_date = earliest_date.replace(day=1,month=1)
+        latest_date = latest_date.replace(day=1,month=1)
+        while earliest_date <= latest_date:
+            time_array.append(earliest_date.strftime("%Y"))
+            if earliest_date in time_dic:
+                temp = time_dic.pop(earliest_date)
+                edits_array.append(temp)
+            else:
+                edits_array.append(0)
+            earliest_date = earliest_date + relativedelta(years= +1)
+
+    data = {
+        'edits_array': edits_array,
+        'time_array': time_array
+    }
+    return JsonResponse(data)
+
+class StatisticsView(View):
+    template_name = 'hashtags/graph.html'
+    form_class = SearchForm
+    def get(self, request):
+        request_dict = request.GET.dict()
+        hashtags = hashtag_queryset(request_dict)
+        context = {'form': self.form_class(request.GET)}
+        context['hashtags'] = hashtags
+        if hashtags:
+            context = get_hashtags_context(request, hashtags, context)
+        return render(request, self.template_name, context=context)
+
+class All_users_view(ListView):
+    template_name = 'hashtags/all_users.html'
+    context_object_name = 'users_list'
+    paginate_by = 30
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(All_users_view, self).get_context_data(**kwargs)
+        request_dict = self.request.GET.dict()
+        hashtags = hashtag_queryset(request_dict)
+        # Context data for stats box
+        context = get_hashtags_context(self.request, hashtags, context)
+        return context
+
+    def get_queryset(self):
+        request_dict = self.request.GET.dict()
+        hashtags = hashtag_queryset(request_dict)
+        users_qs = hashtags.values('username').annotate(edits = Count('rc_id')).order_by('username')
+        return users_qs
+
+class All_projects_view(ListView):
+    template_name = 'hashtags/all_projects.html'
+    context_object_name = 'projects_list'
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(All_projects_view, self).get_context_data(**kwargs)
+        request_dict = self.request.GET.dict()
+        hashtags = hashtag_queryset(request_dict)
+        # Context data for stats box
+        context = get_hashtags_context(self.request, hashtags, context)
+        return context
+
+    def get_queryset(self):
+        request_dict = self.request.GET.dict()
+        hashtags = hashtag_queryset(request_dict)
+        projects_qs = hashtags.values('domain').annotate(edits = Count('rc_id')).order_by('-edits')
+        return projects_qs
+
+def users_csv(request):
```

Comment:
This view (and the top projects one below) duplicates some code from the Json view. It might just be individual lines (e.g. L178, L191) but I think it would be best to move those lines to a helper function so that if we want to edit the Json response the CSV definitely changes in the same way too.

You could even have something like

```
def results_count(qs, field):
  return qs.values(field).annotate(edits = Count('rc_id')).order_by('-edits')
```

so that you could use the same function for both top domains and top usernames.

Reply:
Helper method added